lib/MusicBrainz/Server/Controller/Artist.pm

```diff
@@ -286,26 +286,22 @@ sub show : PathPart('') Chained('load')
     if (defined $base_name) {
         $c->model('Relationship')->load_subset(['artist'], $base_name);
         $c->stash( legal_name => $base_name );
-        my $aliases = $c->model('Artist')->alias->find_by_entity_id($base_name->id);
```

Comment:
Is it safe to assume that `load_related_info` will have saved the aliases to the `Artist` object whenever this is called, or should I include fallback code to try to load aliases if there aren't any?

It seems to work to the extent that I still see legal names when I view artist pages, at least. (Those weren't being shown with this change until I added the `$c->model('Artist')->alias_type->load(@aliases)` call in `load_related_info`.)

Reply:
> Is it safe to assume that load_related_info will have saved the aliases to the Artist object whenever this is called, or should I include fallback code to try to load aliases if there aren't any?

It looks safe to me since `load_related_info` is called in `after 'load'`, while `show` uses `Chained('load')`.

Reply:
It works locally running the beta code, which suggests to me that the product of a prod load without aliases is being cached and causing the beta crash. If that's the case, we could either amend prod to save the info to the artist even if it doesn't use it, or amend beta temporarily to just show nothing rather than ISE if the data is missing. My preference would be the first, if we think it's safe (we're supposed to hotfix prod soon anyway for unrelated reasons)

Reply:
@mwiencek created #3252 to fix this.

Reply:
@mwiencek and @reosarevok, just to make sure I understand this properly, will the caching also cause issues with the `primary_alias` attribute on `Entity::Artist`? The alias is selected based on the UI language. Is that language incorporated into the cache keys, or will there be issues if an entity is cached for a request from a French-using user and then later reused for a request from an English-using user?

Reply:
That shouldn't be an issue as entities are cached immediately before any additional data is loaded onto them. So no `primary_alias` value should be set on the cached copies.